play.pokemonshowdown.com/src/panel-rooms.tsx

```diff
@@ -27,13 +27,27 @@ class RoomsPanel extends PSRoomPanel {
 	static readonly title = "Chat Rooms";
 	hidden = false;
 	search = '';
+	section = '';
 	lastKeyCode = 0;
+	focusedIndex = 0;
+	focusedRoomId = '';
 	override componentDidMount() {
 		super.componentDidMount();
 		this.subscriptions.push(PS.user.subscribe(update => {
 			if (!update && PS.user.named) PS.send(`|/cmd rooms`);
 		}));
 	}
+	override componentDidUpdate() {
+		const el = this.base?.querySelector('a.blocklink.cur');
+		if (!this.focusedIndex) return;
+		if (el) {
+			const rect = el.getBoundingClientRect();
+			const padding = 40;
+			if (rect.top < padding || rect.bottom > window.innerHeight - padding) {
+				el.scrollIntoView({ behavior: 'auto', block: 'nearest' });
+			}
+		}
```

Comment:
```suggestion
		el?.scrollIntoViewIfNeeded({ behavior: 'auto', block: 'nearest' });
```

Reply:
Honestly I would just center it; that's my usual preferred way of dealing with this sort of thing and it's what the teambuilder currently does.